src/services/upload/metadataService.ts

```diff
@@ -149,3 +159,30 @@ export async function parseMetadataJSON(
         // ignore
     }
 }
+
+// tries to extract date from file name if available else returns null
+export function extractDateFromFileName(filename: string): number {
```

Comment:
Shall we put this method instead try catch and return null? Just to unsure that unexpected error won't break upload flow :)